tests/ProjectStorageBackendTest.php

```diff
@@ -90,6 +94,56 @@ public function testProjectStorageAssignBackend(string $backendName): void
         $this->client->purgeDeletedProject($project['id']);
     }
 
+    public function testProjectStorageAssignBackendFailedWithNonNumericBackendId(): void
+    {
+        $maintainer = $this->client->createMaintainer([
+            'name' => 'My test project storage assign maintainer',
+        ]);
+
+        $organization = $this->client->createOrganization($maintainer['id'], [
+            'name' => 'My project storage assign testing organization',
+        ]);
+
+        $project = $this->client->createProject($organization['id'], [
+            'name' => 'My test',
+            'dataRetentionTimeInDays' => 1,
+        ]);
+
+        $client = new GuzzleClient([
+            'base_uri' => getenv('KBC_MANAGE_API_URL'),
+        ]);
+
+        $requestOptions = [
+            'headers' => [
+                'X-KBC-ManageApiToken' => getenv('KBC_MANAGE_API_TOKEN'),
+                'Accept-Encoding' => 'gzip',
+                'Content-Type' => 'application/json',
+                'User-Agent' => 'Keboola Manage API PHP Client',
+            ],
+        ];
+
+        try {
+            $requestOptions['body'] = '{"storageBackendId": "non-numeric"}';
+            $client->post('/manage/projects/' . $project['id'] . '/storage-backend', $requestOptions);
+            $this->fail('Should fail with 400');
+        } catch (GuzzleException $e) {
+            $this->assertEquals(400, $e->getCode());
+            $this->assertStringContainsString('storageBackendId must be an integer.', $e->getMessage());
+        }
+
+        $backend = $this->client->createStorageBackend($this->getSnowflakeBackendCreateOptions());
+
+        // ensure, that backend ID is passed as string into body
+        $requestOptions['body'] = '{"storageBackendId": "'.$backend['id'].'"}';
+        $response = $client->post('/manage/projects/' . $project['id'] . '/storage-backend', $requestOptions);
+        $this->assertEquals(200, $response->getStatusCode());
+
+        $this->client->deleteProject($project['id']);
+        $this->waitForProjectPurge($project['id']);
```

Comment:
ten purge sa da [volat](https://keboolamanagementapi.docs.apiary.io/#reference/deleted-projects/purge-deleted-project/purge-deleted-project) aj s parametrom `ignoreBackendErrors`. Takze rychly fix je pridat do `waitForProjectPurge` parameter `ignoreBackendErrors = false` a v tej metode vo vnutri preposlat ten parameter na https://github.com/keboola/kbc-manage-api-php-client/blob/c5decb33af65a73aec461b2bd61360dab69b37a5/tests/ClientTestCase.php#L344, tu metodu `waitForProjectPurge` pouzivame len v testoch tak by to asi slo ako quick fix, ale chcelo by to opravit asi v connection, ale imho to moze is tako dalsi task

Reply:
opraveno v connection: https://github.com/keboola/connection/pull/4992/commits/1c3345099d9385421c72e0bf4782519282800420

testy fungují